### ISSUE TITLE
[BottomNavigation] Client has no way to set badge text color for A11Y

### ISSUE THREAD

**Issue by romoore** [material-components/material-components-ios#2833]
### Overview
Clients using BottomNavigation should be able to easily set the badge title color (for badge text) to ensure that their choice of badgeColor and text color meet minimum contrast ratio levels (7:1 for small text).

### Reproduction steps
1. Create a BottomNavigationBar
1. Find a way to set the `badgeTitleColor` (or a similar property)

### Version number
44.3.0

---
#### Internal data
- Associated internal bug: [b/117179246](http://b/117179246)

**Comment by andrewoverton** [comment]
When we call `-setItems:` we pass an array of UITabBarItems. In the implementation of that setter each UITabBarItem is turned into a MDCBottomNavigationItemView, outside of the control of the client. Maybe we could give MDCNavigationBar a public `badgeColor` UIColor property that defaults to red and that can be passed to the MDCBottomNavigationItemView class and eventually funnel down to the MDCBottomNavigationItemBadge

**Comment by romoore** [comment]
This was back when we wanted to support UIAppearance. I think now we can just backport [`badgeColor`](https://developer.apple.com/documentation/uikit/uitabbaritem/1648567-badgecolor?language=objc) to iOS 8 and 9?

**Comment by jverkoey** [comment]
@codeman7 Update your status here if still working on it or if you have done some explorations already, otherwise, please unassign.

**Comment by codeman7** [comment]
Added `badgeTextColor` property to `MDCBottomNavigationItemView`. Was suggested in PR to extend UITabBarItem to have `badgeTextColor` property so clients can set the badge text color on all badges individually. After working on that now waiting for design to see if badge text will pass accessibility. If it won't pass then potentially deprecate badges all together. Also, need to confirm with design if we will keep badges because of an unrelated issue.

**Comment by codeman7** [comment]
Research needed is how we will pass accessibility on badge text with the text being so small. If we can't pass accessibility because the text is too small then remove badges all together.

**Comment by romoore** [comment]
@codeman7 Can you clarify what you mean by having the badge text "passing accessibility"? There are no minimum font size requirements, and we simply need to ensure that clients *can* set a badgeColor and badgeTextColor combination that result in a 4:1 contrast ratio.

For example, with a white background for MDCBottomNavigationBar, a Red #700 badge (#D32F2F), and white text, we should get a ratio of 5:1, which is sufficient for WCAG 2.0 AA-level text.

**Comment by ianegordon** [comment]
go/material no longer references badges, nor are they listed as a param in go/material-params.

Can we close this issue as obsolete?

**Comment by romoore** [comment]
Since this is for MDC-iOS and not any internal styling, I think it's best to stick with [material.io guidelines for Bottom Navigation which includes a section on badges](https://material.io/design/components/bottom-navigation.html#behavior). However, since there aren't any specified redlines for badges, we're currently just trying to match UITabBar while providing sufficient parameterization for clients.

**Comment by jverkoey** [comment]
@codeman7 it sounds like your PR was closed because of concerns about a lack of redlines for the new API? In this case it's acceptable for us to move forward with an API even in the absence of designs because the intent of the API is to give clients the ability to tweak colors to meet accessibility guidelines. Whether we have redlines or not can and should be addressed as a separate issue.

Do you feel we can re-open that PR in order to resolve this bug and remove the "Needs research" label?

**Comment by codeman7** [comment]
The PR above I think is on the right track but still doesn't solve the issue.

**Comment by codeman7** [comment]
One alternative is to add an array of UIColor to the component and map the array indexed tabBarItem to the index of the UIColor array. We would have to initialize the array to have all `[UIColor whiteColor]` for the correct amount of items. Then clients would have to make sure they set the correct index of the array.

**Comment by material-mirror** [comment]
The internal issue [b/117179246](http://b/117179246) is now closed. This issue is being closed as a result.

**Comment by codeman7** [comment]
Reopening this issue as clients may want to use a custom `badgeColor` in dark mode which isn't accessible with the white text that the badge currently has.

**Comment by material-mirror** [comment]
The internal issue [b/117179246](http://b/117179246) is now closed. This issue is being closed as a result.
